### PR TITLE
Updating the way Forseti Server Configuration is retrieved from GCS

### PR DESCRIPTION
**Pull request by hiloboy0119**
Moved away from `google_storage_object_signed_url` as it requires
a local json keyfile and I am deploying using service account
impersonation.

https://github.com/terraform-providers/terraform-provider-google/issues/3558

### CODE REVIEW

**Comment by hiloboy0119** [comment]
@googlebot I signed it!

File: examples/on_gke_end_to_end/main.tf [hunk 1]
```diff
@@ -59,6 +59,7 @@ provider "helm" {
   debug                           = true
   automount_service_account_token = true
   install_tiller                  = true
+  version = "~> v0.10"
```
Comment (gkowalski-google) [inline examples/on_gke_end_to_end/main.tf]: This has been added as a version constraint within the other modules in a recent PR, I don't believe this is required anymore.

File: modules/on_gke/main.tf [hunk 2]
```diff
@@ -27,7 +27,7 @@ resource "null_resource" "org_id_and_folder_id_are_both_empty" {
   count = length(var.composite_root_resources) == 0 && var.org_id == "" && var.folder_id == "" ? 1 : 0
 
   provisioner "local-exec" {
-    command     = "echo 'composite_root_resources=${var.composite_root_resources} org_id=${var.org_id} folder_id=${var.org_id}' >&2; false"
+    command     = "echo 'composite_root_resources=${var.composite_root_resources} org_id=${var.org_id} folder_id=${var.folder_id}' >&2; false"
```
Comment (gkowalski-google) [inline modules/on_gke/main.tf]: Nice catch!

File: modules/on_gke/main.tf [hunk 3]
```diff
@@ -124,20 +124,23 @@ data "tls_public_key" "git_sync_public_ssh_key" {
...
+  bucket = module.server_gcs.forseti-server-storage-bucket
+  name   = "configs/forseti_conf_server.yaml"
+  depends_on = [
+    module.server_config.forseti-server-config-md5
+  ]
 }
 
+data "google_client_config" "current" {}
+
 data "http" "server_config_contents" {
```
Comment (gkowalski-google) [inline modules/on_gke/main.tf]: Can you explain how this part works? Sorry I am not familiar with it. Will this cause any breaking changes?

**Comment by gkowalski-google** [comment]
@hiloboy0119 The lint tests failed for 2 reasons:

1. Please run `make docker_generate_docs` to update the docs for the helm_chart_version change
2. [This line](https://github.com/forseti-security/terraform-google-forseti/pull/480/files#diff-8473bd00d76d497515054af758397f3cR62) failed because it's expected that all expressions have the equal signs line up. See my comment on that particular line, which I believe can be removed anyway.

**Comment by gkowalski-google** [comment]
@hiloboy0119 Following up on this PR, please look into my comments above.

**Comment by gkowalski-google** [comment]
@hiloboy0119 Another ping on this PR, do you still wish to move forward with this change?

**Comment by gkowalski-google** [comment]
@hiloboy0119 Checking in to see if you still want to move forward with this PR.

**Comment by gkowalski-google** [comment]
Thanks @hiloboy0119 for this PR. We are now hitting the same issue on [this PR](https://github.com/forseti-security/terraform-google-forseti/pull/541) where I am adding tests for the GKE example. I'm going to merge this change into my branch, so that we can fix the tests and include them in the upcoming release.

**Pull request merged**